src/DADynamoApp/DAEntrypoint.cs

```diff
@@ -0,0 +1,182 @@
+            Console.WriteLine($"Unhandled exception: {e}");
+        }
+
+        public ExternalDBApplicationResult OnStartup(ControlledApplication application)
+        {
+            AppDomain.CurrentDomain.UnhandledException += CurrentDomain_UnhandledException;
+            AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;
+            AppDomain.CurrentDomain.AssemblyResolve += DynamoPathResolver.ResolveAssembly;
+
+            Console.WriteLine("<<!>> Starting to load D4DA");
+
+            try
+            {
+                RevitServices.Transactions.TransactionManager.SetupManager(new RevitServices.Transactions.AutomaticTransactionStrategy());
+                ElementBinder.IsEnabled = true;
+
+                RevitServicesUpdater.Initialize(Updaters);
+
+                DesignAutomationBridge.DesignAutomationReadyEvent += HandleDesignAutomationReadyEvent;
+
+                Console.WriteLine("<<!>> D4DA Loaded");
+
+                return ExternalDBApplicationResult.Succeeded;
+            }
+            catch (Exception ex)
+            {
+                return ExternalDBApplicationResult.Failed;
+            }
+        }
+
+        public void HandleDesignAutomationReadyEvent(object sender, DesignAutomationReadyEventArgs e)
+        {
+            Console.WriteLine("<<!>> DA event raised.");
+            WorkItemFolder = Directory.GetCurrentDirectory();
+
+            foreach (var xx in Directory.EnumerateFiles(WorkItemFolder))
+                Console.WriteLine(xx);
+
+            var dynTempDir = Path.Combine(WorkItemFolder, "dyn_tmp");
+            Console.WriteLine($"<<!>> Work folder is '{WorkItemFolder}'");
+            var app = e.DesignAutomationData?.RevitApp;
+            Console.WriteLine("<<!>> Preparing Dynamo model. Vers 1");
+
+            var hostloc = typeof(Autodesk.Revit.ApplicationServices.Application).Assembly.Location;
+            var asmLocation = Path.GetDirectoryName(hostloc);
+            Console.WriteLine($"using asm at location {asmLocation}");
+            Console.WriteLine($"Is Loaded {LoadMessage}");
+            var curFolder = Directory.GetParent(Assembly.GetExecutingAssembly().Location).FullName;
+            var cmdArgs = new Dynamo.Applications.StartupUtils.CommandLineArguments();
+            cmdArgs.ASMPath = asmLocation;
+            cmdArgs.DisableAnalytics = true;
+            cmdArgs.ServiceMode = true;
+            cmdArgs.AnalyticsInfo = new() { HostName = "Revit_DA" };
+            cmdArgs.ImportedPaths = [Path.Combine(curFolder, "nodes", "RevitNodes.dll"), Path.Combine(curFolder, "nodes", "DSRevitNodesUI.dll")];
+            // need this for cloud, does not work on local
+            cmdArgs.UserDataFolder = Path.Combine(dynTempDir, "Dynamo Revit");
+            cmdArgs.CommonDataFolder = Path.Combine(dynTempDir, "Dynamo");
+
+            // Startup a new project, maybe an option we can have ?
+            //app.NewProjectDocument(Autodesk.Revit.DB.UnitSystem.Metric);
```

Comment:
If I were a user executing against dynamo as a service - I would expect this information to somehow have been supplied via a web UI or via part of the graph authoring experience ahead of time.

For example:

1. I authored the graph in a metric project, so that should be taken into account when run in this context against another project.
2. I set this graph to run against a particular project already so we should just open that project.


Do you have a user use case in mind where a graph should be run on a brand new project?